dipy/align/tests/test_imaffine.py

```diff
@@ -312,7 +320,232 @@ def test_mi_gradient():
+
+    Returns
+    -------
+    transforms: sequence of (dim + 1)x(dim + 1) matrices
+        each matrix correspond to an affine transform with a combination
+        of the input parameters
+    """
+    transforms = []
+    for t in translations:
+        trans_inv = np.eye(dim + 1)
+        trans_inv[:dim, dim] = -t[:dim]
+        trans = npl.inv(trans_inv)
+        for theta in rotations:  # rotation angle
+            if dim == 2:
+                ct = np.cos(theta)
+                st = np.sin(theta)
+                rot = np.array([[ct, -st, 0],
+                                [st, ct, 0],
+                                [0, 0, 1]])
+            else:
+                rot = np.eye(dim + 1)
+                rot[:3, :3] = geometry.rodrigues_axis_rotation(rot_axis, theta)
+
+            for s in scales:  # scale
+                scale = np.eye(dim + 1) * s
+                scale[dim,dim] = 1
+
+            affine = trans.dot(scale.dot(rot.dot(trans_inv)))
+            transforms.append(affine)
+    return transforms
+
+
+def test_affine_map():
+    np.random.seed(2112927)
+    dom_shape = np.array([64, 64, 64], dtype=np.int32)
+    cod_shape = np.array([80, 80, 80], dtype=np.int32)
+    nx = dom_shape[0]
+    ny = dom_shape[1]
+    nz = dom_shape[2]
+    # Radius of the circle/sphere (testing image)
+    radius = 16
+    # Rotation axis (used for 3D transforms only)
+    rot_axis = np.array([.5, 2.0, 1.5])
+    # Arbitrary transform parameters
+    t = 0.15
+    rotations = [-1 * np.pi / 10.0, 0.0, np.pi / 10.0]
+    scales = [0.9,  1.0, 1.1]
+    for dim in [2, 3]:
+        # Setup current dimension
+        if dim == 2:
+            # Create image of a circle
+            img = vf.create_circle(cod_shape[0], cod_shape[1], radius)
+            oracle_linear = vf.transform_2d_affine
+            oracle_nn = vf.transform_2d_affine_nn
+        else:
+            # Create image of a sphere
+            img = vf.create_sphere(cod_shape[0], cod_shape[1], cod_shape[2],
+                                   radius)
+            oracle_linear = vf.transform_3d_affine
+            oracle_nn = vf.transform_3d_affine_nn
+        img = np.array(img)
+        # Translation is the only parameter differing for 2D and 3D
+        translations = [t * dom_shape[:dim]]
+        # Generate affine transforms
+        gt_affines = create_affine_transforms(dim, translations, rotations,
+                                              scales, rot_axis)
+        # Include the None case
+        gt_affines.append(None)
+
+        for affine in gt_affines:
+
+            # make both domain point to the same physical region
+            # It's ok to use the same transform, we just want to test
+            # that this information is actually being considered
+            domain_grid2world = affine
+            codomain_grid2world = affine
+            grid2grid_transform = affine
+
+            # Evaluate the transform with vector_fields module (already tested)
```

Comment:
Yes, exactly, `transform_2d_affine` and `transform_3d_affine` have their own tests, which compare against scipy's `map_coordinates`. In this case, the test is much cleaner by using `transform_2d_affine` and `transform_3d_affine` as oracles instead of creating the full coordinate arrays to call `map_coordinates`.